.github/workflows/tests.yml

```diff
@@ -75,38 +80,40 @@ jobs:
           python-version: ${{ matrix.python-version}}
           allow-prereleases: true
 
-      - name: Build package
+      - name: Install build dependencies
+        if: ${{ matrix.INSTALL_FROM_SDIST != 1 }}
+        env:
+          PIP_FLAGS: ${{ matrix.PIP_FLAGS }}
+        run: |
+          source tools/github/setup_build_env.sh
+
+      - name: Build and install from source
+        if: ${{ matrix.INSTALL_FROM_SDIST != 1 }}
+        run: |
+          pip install -vv --no-build-isolation .
```

Comment:
I'm wondering now why we actually use `--no-build-isolation` at all. Seems to me that we could remove the "Install build dependencies" step before and simply rely on pip install to do its thing? 

Might be something we could look into as a follow-up simplification. But not terribly important.

Reply:
You actually set it up this way already for `INSTALL_FROM_SDIST` which skips installing the builds and uses an isolated build env.

Reply:
I suppose it shortens the subsequent testing and documentation dependency installation steps, but it's not a big deal; we could, indeed, remove that step.